Automattic/wp-calypso
# Subscription management: queue resubscribe to fire after unsubscribe succeeded

Closes #78626

## Proposed Changes

This PR changes the behavior of the unsubscribe toast. Clicking Resubscribe will now only fire after the Unsubscribe mutation has succeeded.

`useRef` is used instead of `useState` because the callback function would otherwise capture the state at the moment of invocation.

## Testing Instructions

- Apply this PR
- Go to `http://calypso.localhost:3000/subscriptions/sites`
- Unsubscribe from a site & resubscribe quickly
- Refresh, the site should still be there

To more easily test this, add this before the API call in `packages/data-stores/src/reader/mutations/use-site-unsubscribe-mutation.ts` (line 42):

`await new Promise(r => setTimeout(r, 10000));`

This will introduce a 10s timeout.

## Pre-merge Checklist
